src/service/PetClinicService.ts

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+
+class PetService {
+  private url: string = 'http://localhost:9966/petclinic/api';
```

Comment:
While this is technically no problem to use a service here, I think in React applications this is a rather uncommon pattern, so I would suggest that we avoid it here:
- you can simple replace the class by single named exports of your find methods 
  - then you also don't have the problem with the creation of the instance in the render cycle of the components
- I would prefer using [Tanstack Query](https://tanstack.com/query/latest) instead as this is a more "modern" approach these days (in best case combined with [Zod](https://github.com/colinhacks/zod) 😉) and solves typical React problems when working with data fetching (like effects) and provides caching etc.

Reply:
thanks for your feedback! I've replaced the class by single named exports as you mentioned. 

Regarding Tanstack query and Zod: if that's ok with you, we can create a separate issue and do it once the migration will be over. As we're still focusing on the basics for now :)